### PR TITLE
(CONT-811) Add Ruby 3.2 support

### PR DESCRIPTION
**Pull request by GSPatton**
In preparation for the release of Puppet 8, this PR adds Ruby 3.2 support and removes Ruby 2.5 for puppet-lint.

### CODE REVIEW

File: puppet-lint.gemspec [hunk 1]
```diff
@@ -33,4 +33,5 @@ Gem::Specification.new do |spec|
   spec.license = 'MIT'
 
   spec.required_ruby_version = Gem::Requirement.new('>= 2.5'.freeze)
+  spec.metadata['rubygems_mfa_required'] = 'true'
```
Comment (ekohl) [inline puppet-lint.gemspec:36]: How do you release with MFA via GHA? I see you have https://github.com/puppetlabs/puppet-lint/blob/main/.github/workflows/release.yml and I question how you achieve this.

File: lib/puppet-lint/optparser.rb [hunk 2]
```diff
@@ -143,7 +143,7 @@ def self.build(args = [])
 
     unless args.include?('--no-config')
       opt_parser.load('/etc/puppet-lint.rc')
-      if ENV.key?('HOME') && File.readable?(ENV['HOME'])
+      if ENV.key?('HOME') && File.readable?(Dir.home)
```
Comment (ekohl) [inline lib/puppet-lint/optparser.rb:146]: TIL: `Dir.home` exists. I think the `ENV.key?('HOME')` check no longer makes sense if you use `Dir.home`.

File: lib/puppet-lint/monkeypatches.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@
 rescue
   # monkeypatch String#% into Ruby 1.8.7
```
Comment (ekohl) [inline lib/puppet-lint/monkeypatches.rb:4]: Given the lower bound is now set at 2.5, I wonder how much of this could be dropped.

Comment (GSPatton) [reply lib/puppet-lint/monkeypatches.rb:4]: yes I've wondered about this file. I'll discuss this with the team

Comment (GSPatton) [reply lib/puppet-lint/monkeypatches.rb:4]: hey @ekohl. I had a look at this file and am not entirely sure what its purpose is and so don't feel entirely confident to make any changes to it. In the interest of moving on with adding ruby 3 support across our tools I'm going to get this PR merged but add a ticket to our backlog to investigate this more. Thanks for highlighting this for us.

**Pull request merged**